script.js

```diff
@@ -1,4 +1,165 @@
+//Global Variable
+var username =0
+var computerWin = 0
+var userWin = 0
+var roundCount = 0
+var gamemode = 0
+var firstWinner = 0     //0 = no lead, 1 = User lead, 2 = computer lead
+var ultiWinner = 0
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // input username
+  if (username == 0){
+    username = input
+    return "Hi "+username+" :) <br> Please select game mode  (normal/reversed/korean/computer)"
+  } else{   
+  // select game mode
+  if (gamemode == 0 && input == "normal"){
+    gamemode = input
+    return "Normal mode selected. Please enter scissors paper or stone"
+  }else if (gamemode == 0 && input == "reversed"){
+    gamemode = input
+    return "Reversed mode selected. Please enter scissors paper or stone"
+  }else if (gamemode == 0 && input == "korean"){
+      gamemode = input
+      return "Korean mode selected. Please enter scissors paper or stone"
+  }else if (gamemode == 0 && input == "computer"){
+      gamemode = input
+      return "Computer mode selected. Computer will choose your input, please press submit again"
+  }else if (gamemode == 0 && (input != "normal" || input != "reversed")){
+    return "Hi "+username+" :) <br> Please select game mode  (normal/reversed/korean/computer)"
+  }
+  }
+
+  //Computer Version
+  if (gamemode == "computer"){
+    programHand = chooseHand();
+    input = chooseHand();
+    gameStatus = checkWin(input, programHand);
+    return (
+      "You selected "+ gamemode +" mode"+
+      "<br>Computer selected " +
+      programHand +
+      "<br>" +username+" selected " +
+      input +
+      "<br><br>" +
+      gameStatus)
+  }
+
+  //korean version
```

Comment:
well done implementing the game logic for the korean mode and the ability to switch modes!